### ISSUE TITLE
Ubuntu home directory not saved when resetting Windows 10

### ISSUE THREAD

**Issue by xobs** [Microsoft/WSL#2970]
* Your Windows build number:  10.0.16299.125

* What you're doing and what's happening:  When doing a "Reset" of Windows 10, the Ubuntu home directory is erased

* What's wrong / what should be happening instead: The system advertises that "Your files and folders will be preserved", however the entirety of the Ubuntu home directory is removed instead, along with the root filesystem.

It is understandable that the act of reinstalling an operating system would cause applications to be uninstalled, and give that Ubuntu is considered an application it is understandable that Ubuntu is uninstalled.  However, the fact that the process advertises that "Your files and folders will be preserved" when in fact the entire root filesystem (including home directory) is erased is somewhat startling.

The package's LocalState directory is created inside the \Windows.old\Users\...\ directory, however it is empty.

It should be communicated that users should avoid storing important files inside LXSS, and the installer shouldn't advertise that "Files and folders will be preserved".  Even better, it would be nice if the LXSS files were backed up as part of the process.

**Comment by Biswa96** [comment]
If you download it from Windows Store it stores Ubuntu rootfs in `C:\Users\%USERNAME%\AppData\Local\Packages\`. If that folder deleted the whole Ubuntu system will be removed.

**Comment by xobs** [comment]
Correct.  The Windows 10 "Reset my PC" function moves C:\Users into C:\Windows.old\Users, but doesn't move the Ubuntu rootfs.  In fact, it doesn't appear to copy much of the contents of AppData at all.  As a result, user data is deleted despite the "Keep my files", including the entire home folder.

**Comment by therealkenc** [comment]
> "Your files and folders will be preserved" 

Windows doesn't know where your home directory resides; indeed _WSL proper_ doesn't know where it resides, because you can `useradd -d /allmystuff/me/`. I'll see if anyone weighs in, but this is appears pretty by-design. See for example [this](https://superuser.com/a/1097399). That message you cite could also warn that your high scores for some Store-installed game might be blown away. I mean, it could warn about that. Windows Feedback Hub, or something. But they can't make the message specific to WSL, because they don't know you've installed Ubuntu anymore than they know you've installed Candy Crush.

I think mostly this goes to a lack of `mount -t drvfs`, though I might be projecting a little. Then you could put your "My WSL Neighbourhood" beside "My Pictures" and "My Music". 

> The Windows 10 "Reset my PC" function moves C:\Users into C:\Windows.old\Users, but doesn't move the Ubuntu rootfs

That's a little different though. **If** the AppData from your **other** Store applications are being moved into `Windows.old` **but not** Ubuntu, that would be weird. _If_ on the other hand **none** of `AppData` gets backed up from your Store apps, then maybe it should. But per first paragraph that wouldn't be a WSL thing.

**Comment by xobs** [comment]
I agree that it's working "As Designed", and the issue is that the design is somewhat surprising.

I think it's fine that the root filesystem isn't preserved -- that's mostly what happens with a `Reset my PC` function anyway.  What I don't think is fine is the home directory being erased.

What about making /home a symlink to a subdirectory in the user's `Documents`?  Is there an issue with me, as a user, symlinking `Documents`, `Music`, etc. into my Ubuntu home directory and operating from there?  I see warnings not to interact with Linux files from Windows, and with this issue I'm more inclined to ignore those warnings.  What are the repercussions of doing so?

**Comment by therealkenc** [comment]
You are on 16299, which doesn't have the DrvFS metadata improvements in [17063](https://docs.microsoft.com/en-us/windows/wsl/release-notes#build-17063) (and more onward). So you'll face some DrvFs limitations in your WSL version, which you might or might not care about depending on what you're doing. 

Those [warnings](https://blogs.msdn.microsoft.com/commandline/2016/11/17/do-not-change-linux-files-using-windows-apps-and-tools/) are regarding mucking around in the WSL filesystem (LxFs), not `/mnt/c/Users/you/somewhere` (DrvFs). DrvFs is fair game. But usually the motivator for storing your files in DrvFs is more fundamental than worrying about them being blown away by a Windows reset.  If you keep your files on DrvFs you can access them with Windows editors and other tools. It isn't how I roll personally, but it is a supported scenario.

**Comment by dezza** [comment]
It would really be nice if Microsoft considered installed WSL distributions "personal files" as well 👍 I'd argue that implementing just saving them when resetting (and choosing 'Keep Personal Files') would be easier, than make assumptions about what files to save (e.g. home directory as suggested).

Maybe there could be implications with starting it again after, not familiar with how WSL is registered against Windows. Definetily better, if WSL was considered "personal files"
